test/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,564 @@
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private static void test_triggerDumpToFile(OpenJ9DiagnosticsMXBean diagBean, String test) {
+		boolean found = false;
+		String[] dumpAgents = {"java", "heap", "snap", "system", "stack"};
+		String[] files = {"javacore.%Y%m%d.%H%M%S", "heapdump.%H%M%S", "Snap.%pid.trc", "core.%pid.dmp", "stack.%seq.txt"};
+		int index = 0;
+		String dir = "." + File.separator + test + "_dumps";
+
+		for (String agent : dumpAgents) {
+			try {
+				String filePath = dir + File.separator + files[index];
+
+				String fileName = diagBean.triggerDumpToFile(agent, filePath);
+				File dumpFile = new File(fileName);
+
+				if (!(agent.equals("stack"))) {
+					found = findAndDeleteFile(dir, dumpFile.getName());
+					Assert.assertTrue(found, fileName + " not found");
+					
+				} 
+			
+			} catch (java.lang.IllegalArgumentException e) {
+				/* stack agent is not supported by the trigger method */
+				if (agent.equals("stack")) {
+					Assert.assertTrue(true, "Did not receive the expected exception IllegalArgumentException");
+				} 
+				else {
+					Assert.fail("Unexpected IllegalArgumentException : " + e.getMessage());
+				}
+			} catch (Exception e) {
+				Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+				e.printStackTrace();
+			}
+
+			index++;
+		}
+	}
+
+	/**
+	 * Function to test if a class heap dump is created
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private static void test_triggerClassicHeapDump(OpenJ9DiagnosticsMXBean diagBean) {
+		boolean found = false;
+
+		try {
+			String fileName = diagBean.triggerClassicHeapDump();
+			found = findAndDeleteFile(".", new File(fileName).getName());
+			Assert.assertTrue(found, fileName + " not found");
+		} catch (InvalidDumpOptionException e){
+			Assert.fail("Unexpected InvalidDumpOptionException : " + e.getMessage());
+		} catch (Exception e){
+			Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+			e.printStackTrace();
+		}
+
+	}	
+
+
+	/**
+	 * Function to test if the dump options were reset to what it was at JVM initialization
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private void test_resetDumpOptions(OpenJ9DiagnosticsMXBean diagBean) {
+		boolean found = false;
+		try {
+			diagBean.setDumpOptions("java+heap+system:events=vmstop");			
+		} catch (InvalidDumpOptionException e) {
+			Assert.fail("Unexpected InvalidDumpOptionException : " + e.getMessage());
+		} catch (Exception e) {
+			Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+		}		
+		
+	
+		try {	
+			diagBean.resetDumpOptions();
+		} catch (DumpConfigurationUnavailableException e) {
+			Assert.fail("Unexpected DumpConfigurationUnavailableException : " + e.getMessage());
+		}
+
+		List<String> newDumpOptions = new ArrayList<String>();
+		newDumpOptions = getDumpOptions();
```

Comment:
Please remove the first assignment to 'newDumpOptions', just initialize with 'getDumpOptions()'.